### ISSUE TITLE
[Chore]: Loosen Job id regex restrictions to not require GUIDs

### ISSUE THREAD

**Issue by tcdowney** [cloudfoundry/korifi#1116]
### Background

`kubectl` users don't have to make `metadata.name` a GUID for their resources (e.g. `CFOrg`, `CFSpace`). They just need to make something unique.

This works fine for most endpoints, but the `Jobs` endpoint validate that the `Job` identifier we send through conforms to a strict structure.

### Action to take

Can we eliminate the strict format we have in place here: https://github.com/cloudfoundry/korifi/blob/78e3d0a5bd88ec276b2d867a2962e7b64d8daa14/api/apis/job_handler.go#L74

We still need something deterministic that we can split on for the "job type" portion of the `Job` id, but the second part should be able to accept any valid `metadata.name`. Maybe we should pick an identifier that can't be part of a K8s `metadata.name` to be the delimiter between type and resource id.

### Impact

This will allow `kubectl` users to create resources without needing to be aware of and having to abide to a strict naming convention.

### Dev Notes

- Kubernetes resource name restrictions: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
  - ^^ For choosing a delimiter that is not a valid `metadata.name` character.

**Comment by tcdowney** [comment]
One option for a delimiter is `~`. It's valid, is not used by Cloud Controller as a job `operation` type, and has no reserved use in a URI.

```
2.3.  Unreserved Characters

   Characters that are allowed in a URI but do not have a reserved
   purpose are called unreserved.  These include uppercase and lowercase
   letters, decimal digits, hyphen, period, underscore, and tilde.

      unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
```

https://www.ietf.org/rfc/rfc3986.txt

**Comment by tcdowney** [comment]
This was addressed in #1156. Acceptance steps:

## Validating Existing Behavior
The commands that use CF `Jobs` continue to work with the new job identifiers:

### Pushing an app (space manifest apply)
```
cf push node -v | grep Location
Location: https://api.k8s.tim.example.com/v3/jobs/space.apply_manifest~cf-space-161b269b-732c-4bff-9a6e-ac9288caa435
```

### Pushing an app (space manifest apply)
```
$ cf push node -v | grep Location
Location: https://api.k8s.tim.example.com/v3/jobs/space.apply_manifest~cf-space-161b269b-732c-4bff-9a6e-ac9288caa435
```

### Deleting an app
```
$ cf delete node -v -f | grep Location
Location: https://api.k8s.tim.example.com/v3/jobs/app.delete~5376cf03-2933-411f-9e26-8bd26fc59fb5
```

### Delete a route
```
$ cf delete-route -v k8s.tim.migration.quest -n node -f | grep Location
Location: https://api.k8s.tim.example.com/v3/jobs/route.delete~cf-route-d65b7ad8-5690-42f2-a629-e8ef42e14858
```

### Delete a space
```
$ cf delete-space s -f -v | grep Location
Location: https://api.k8s.tim.example.com/v3/jobs/space.delete~cf-space-161b269b-732c-4bff-9a6e-ac9288caa435
```

### Delete an org
```
$ cf delete-org o -f -v | grep Location
Location: https://api.k8s.tim.example.com/v3/jobs/org.delete~cf-org-82287c83-245c-46cb-acf2-9ec28b99e481
```

## Custom CRD shenanigans
Manually applied the following `CFSpace` CRD with `kubectl`:
```yaml
---
apiVersion: korifi.cloudfoundry.org/v1alpha1
kind: CFSpace
metadata:
  name: tim-space
  namespace: cf-org-a2f46864-941b-4bb3-b75c-14978fb020f8
spec:
  displayName: tims-custom-space
```

The namespace didn't apply directly, turns out we need to set a `cf` prefix on them for HNC. 
```
1.6541894458154516e+09	ERROR	controllers.CFSpace	Error creating SubnamespaceAnchor for CFSpace cf-org-a2f46864-941b-4bb3-b75c-14978fb020f8/tim-space	{"error": "admission webhook \"subnamespaceanchors.hnc.x-k8s.io\" denied the request: subnamespaceanchors.hnc.x-k8s.io \"tim-space\" is forbidden: cannot create a subnamespace using the unmanaged namespace name \"tim-space\" (does not match the regex set by the HNC administrator: `^cf.*$`)"}
```

So applied:

```yaml
---
apiVersion: korifi.cloudfoundry.org/v1alpha1
kind: CFSpace
metadata:
  name: cf-tim-space
  namespace: cf-org-a2f46864-941b-4bb3-b75c-14978fb020f8
spec:
  displayName: tims-custom-space
```

And that worked. I'm not too concerned about requiring the `cf` prefix since we've got active work in flight around removing HNC (https://github.com/cloudfoundry/korifi/pull/1167).

So then I did a `cf push` which is what failed previously:
```
$ cf push tim-node
Pushing app tim-node to org my-custom-org / space tims-custom-space as cf-admin...
Packaging files to upload...
Uploading files...
 1017 B / 1017 B [==================================================================================================================================================] 100.00% 1s

Waiting for API to complete processing files...

Staging app and tracing logs...

Waiting for app tim-node to start...

Instances starting...
Instances starting...

name:              tim-node
requested state:   started
routes:            tim-node.k8s.tim.migration.quest
last uploaded:     Thu 02 Jun 11:16:42 MDT 2022
stack:             io.buildpacks.stacks.bionic
buildpacks:

type:            web
sidecars:
instances:       1/1
memory usage:    1024M
start command:   node /workspace/server.js
     state     since                  cpu    memory   disk     details
#0   running   2022-06-02T17:17:24Z   0.0%   0 of 0   0 of 0
```

Worked well! Then `$ cf delete-space tims-custom-space` worked to remove it.

**Comment by tcdowney** [comment]
Marking this chore as complete.
